controllers/cinderapi_controller.go

```diff
@@ -256,18 +256,20 @@ func (r *CinderAPIReconciler) reconcileDelete(ctx context.Context, instance *cin
 
 			ks, err := keystone.GetKeystoneServiceWithName(ctx, helper, ksSvc["name"], instance.Namespace)
 
-			if err != nil {
-				if k8s_errors.IsNotFound(err) {
-					return ctrl.Result{}, nil
-				}
+			if err != nil && !k8s_errors.IsNotFound(err) {
 				return ctrl.Result{}, err
 			}
 
-			ksSvcObj := keystone.NewKeystoneService(ks.Spec, instance.Namespace, map[string]string{}, 10)
-			err = ksSvcObj.Delete(ctx, helper)
+			// Could get here if there was an error and it was "not found", so need to check
+			// (we do it this way because we want to make sure the "RemoveFinalizer" call is
+			// executed even if the KeystoneService no longer exists)
+			if err == nil && ks != nil {
```

Comment:
we'd have to also change https://github.com/openstack-k8s-operators/keystone-operator/blob/master/pkg/external/funcs.go#L247 to return nil for the ks if not found.

Reply:
Ah yeah, forgot about that.  I'll do that in another PR now.

Reply:
Actually, for the moment, let's just check here for `ks != nil` only.  The required changes to the Keystone operator have ripple effects that I want to make sure I understand.